pkg/typesystem/typesystem.go

```diff
@@ -108,45 +78,65 @@ func (t *TypeSystem) GetRelation(objectType, relation string) (*openfgapb.Relati
 	return r, nil
 }
 
-// Validate validates the type system according to the following rules:
+// Validate validates an *openfgapb.AuthorizationModel according to the following rules:
 //  1. For every rewrite the relations in the rewrite must:
-//     a. Be valid relations on the same type in the authorization typeSystem (in cases of computedUserset)
+//     a. Be valid relations on the same type in the authorization model (in cases of computedUserset)
 //     b. Be valid relations on another existing type (in cases of tupleToUserset)
-//  2. Do not allow duplicate types or duplicate relations (but that is inherent in the map structure so nothing to
-//     actually check)
+//  2. Do not allow duplicate types or duplicate relations (only need to check types as relations are
+//     in a map so cannot contain duplicates)
 //
-// If it is a SchemaVersion1_1 type system (with types on relations), then additionally validate the type system
+// If it is a 1.1 authorization model (with types on relations), then additionally validate the type system
 // according to the following rules:
 //  3. Every type on a relation must be a valid type:
```

Comment:
Every type restriction on a relation must be a valid type

Reply:
Done. Thanks!